app/domain/location/service.js

```diff
@@ -3,38 +3,67 @@
 angular.module('openhds')
 
     .service('LocationService',
-        ['$rootScope', '$http',
-            function($rootScope, $http) {
-                var service = this;
-                var headers;
-
-
-
-                function Request(model) {
-                    return {
-                        collectedByUuid: model.currentFieldworker.uuid,
-                        locationHierarchyUuid: model.currentHierarchy.uuid,
-                        location: {
-                            name: model.location.name,
-                            extId: model.location.extId,
-                            type: model.location.type,
-                            collectionDateTime: model.collectionDateTime
-                        }
-                    };
-                }
-
-                service.submit = function (model, callback) {
-                    headers = {
-                        headers: {
-                            authorization: "Basic " + $rootScope.credentials
-                        }
-                    };
-                    var url = $rootScope.restApiUrl + "/locations";
-                    var request = Request(model);
-                    $http.post(url, request, headers).then(function (response) {
-                        callback(response.data);
-                    });
-                };
-
-                return service;
-            }]);
+             ['$rootScope', '$http', '$q',
+              function($rootScope, $http, $q) {
+                  var service = this;
+                  var headers;
+
+
+
+                  function Request(model) {
+                      return {
+                          collectedByUuid: model.currentFieldworker.uuid,
+                          locationHierarchyUuid: model.currentHierarchy.uuid,
+                          location: {
+                              name: model.location.name,
+                              extId: model.location.extId,
+                              type: model.location.type,
+                              collectionDateTime: model.collectionDateTime
+                          }
+                      };
+                  }
+
+                  function getHeaders() {
+                      return {
+                          headers: {
+                              authorization: "Basic " + $rootScope.credentials
+                          }
+                      };
+                  }
+
+                  service.getByHierarchy = function(hierarchyUuid) {
```

Comment:
you can use this as

    svc.getByHierarchy(uuid).then(function(response) {
        vm.locationsByHierarchy = response;
    }


It returns a list of 

    {description: loc.description,
     extId: loc.extId,
     type: loc.type,
     uuid: loc.uuid,
     name: loc.name
     }